tests/unit/modules/test_nxos.py

```diff
@@ -0,0 +1,1024 @@
+
+                        # Execute the function under test
+                        result = nxos.config(commands, **kwargs)
+                        self.assertEqual(result, expected_output)
+
+    def test_config_file(self):
+
+        """ UT: nxos module:config method - Using config_file arg"""
+
+        config_file = 'salt://bgp_config.txt'
+        kwargs = {}
+        expected_output = 'COMMAND_LIST: feature bgp ; ! ; router bgp 55 ; address-family ipv4 unicast ; no client-to-client reflection ; additional-paths send\n\n'
+
+        with patch.object(nxos, 'show', MagicMock(return_value=initial_config_file)):
+            mock_cmd = MagicMock(return_value=config_input_file)
+            with patch.dict(nxos.__salt__, {'cp.get_file_str': mock_cmd}):
+                mock_cmd = MagicMock(return_value=template_engine_file_str_file)
+                with patch.dict(nxos.__salt__, {'file.apply_template_on_contents': mock_cmd}):
+                    with patch.object(nxos, '_configure_device', MagicMock(return_value=config_result_file)):
+                        with patch.object(nxos, 'show', MagicMock(return_value=modified_config_file)):
+
+                            # Execute the function under test
+                            result = nxos.config(config_file=config_file, **kwargs)
+                            self.assertEqual(result, expected_output)
+
+    def test_config_file_error1(self):
+
+        """ UT: nxos module:config method - Error file not found """
+
+        config_file = 'salt://bgp_config.txt'
+        kwargs = {}
+
+        with patch.object(nxos, 'show', MagicMock(return_value=initial_config_file)):
+            mock_cmd = MagicMock(return_value=False)
+            with patch.dict(nxos.__salt__, {'cp.get_file_str': mock_cmd}):
+                mock_cmd = MagicMock(return_value=template_engine_file_str_file)
+                with patch.dict(nxos.__salt__, {'file.apply_template_on_contents': mock_cmd}):
+                    with patch.object(nxos, '_configure_device', MagicMock(return_value=config_result_file)):
+                        with patch.object(nxos, 'show', MagicMock(return_value=modified_config_file)):
+
+                            # Execute the function under test
+                            with self.assertRaises(CommandExecutionError):
+                                nxos.config(config_file=config_file, **kwargs)
+
+    def test_config_nxos_error_ssh(self):
+
+        """ UT: nxos module:config method - nxos device error over ssh transport """
+
+        commands = ['feature bgp', 'router bgp 57']
+        kwargs = {}
+        config_result = [['feature bgp', 'router bgp 57'], u'bgp instance is already running; Tag is 55']
+        expected_output = 'COMMAND_LIST: feature bgp ; router bgp 57\nbgp instance is already running; Tag is 55\n'
+
+        with patch.object(nxos, 'show', MagicMock(return_value=initial_config[0])):
```

Comment:
we're patching 'show' twice, we should change this to be `side_effect = [initial_config[0], modified_config[0]]`

This applies to several of these tests.

Reply:
@mikewiebe, could you make this change?

Reply:
Yes, I can make this change.

Reply:
(these are the currently failing tests)